compiler/noirc_frontend/src/tests.rs

```diff
@@ -3623,3 +3623,15 @@ fn use_type_alias_to_generic_concrete_type_in_method_call() {
     "#;
     assert_no_errors(src);
 }
+
+#[test]
+fn allows_struct_with_generic_infix_type_as_main_input() {
+    let src = r#"
+        struct Foo<let N: u32> {
+            x: [u64; N * 2],
+        }
+
+        fn main(_x: Foo<18>) {}
```

Comment:
Does this work with cases e.g. `fn main(_x: Foo<2 * 9>)` as well?

Reply:
Or `(_x: Foo<N * 2>)` where `N` is a constant global?

Reply:
It seems both work. Should I add tests for those too?

Reply:
Yeah, more tests never hurts.

Reply:
Done!